### ISSUE TITLE
Install Issue

### ISSUE THREAD

**Issue by suchidog** [thorrak/fermentrack#398]
Hi,

I am trying to install Fermentrack on a RaspberryPi 4 and keep getting an install error as follows:

Building dependency tree...
Reading state information...
build-essential is already the newest version (12.6).
git is already the newest version (1:2.20.1-2).
The following additional packages will be installed:
  libftdi1 libhidapi-libusb0 libhiredis0.14 libjemalloc2 liblua5.1-0
  libnginx-mod-http-auth-pam libnginx-mod-http-dav-ext libnginx-mod-http-echo
  libnginx-mod-http-geoip libnginx-mod-http-image-filter
  libnginx-mod-http-subs-filter libnginx-mod-http-upstream-fair
  libnginx-mod-http-xslt-filter libnginx-mod-mail libnginx-mod-stream
  libusb-0.1-4 lua-bitop lua-cjson nginx-common nginx-full redis-tools
Suggested packages:
  dfu-programmer avrdude-doc fcgiwrap nginx-doc ssl-cert ruby-redis
The following NEW packages will be installed:
  avrdude libftdi1 libhidapi-libusb0 libhiredis0.14 libjemalloc2 liblua5.1-0
  libnginx-mod-http-auth-pam libnginx-mod-http-dav-ext libnginx-mod-http-echo
  libnginx-mod-http-geoip libnginx-mod-http-image-filter
  libnginx-mod-http-subs-filter libnginx-mod-http-upstream-fair
  libnginx-mod-http-xslt-filter libnginx-mod-mail libnginx-mod-stream
  libusb-0.1-4 lua-bitop lua-cjson nginx nginx-common nginx-full redis-server
  redis-tools
0 upgraded, 24 newly installed, 0 to remove and 130 not upgraded.
Need to get 2,197 kB/2,834 kB of archives.
After this operation, 7,263 kB of additional disk space will be used.
Err:1 http://raspbian.raspberrypi.org/raspbian buster/main armhf redis-tools armhf 5:5.0.3-4
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:2 http://raspbian.raspberrypi.org/raspbian buster/main armhf redis-server armhf 5:5.0.3-4
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:3 http://raspbian.raspberrypi.org/raspbian buster/main armhf nginx-common all 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:4 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-auth-pam armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:5 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-dav-ext armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:6 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-echo armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:7 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-geoip armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:8 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-image-filter armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:9 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-subs-filter armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:10 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-upstream-fair armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:11 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-http-xslt-filter armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:12 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-mail armhf 1.14.2-2
 404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:13 http://raspbian.raspberrypi.org/raspbian buster/main armhf libnginx-mod-stream armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:14 http://raspbian.raspberrypi.org/raspbian buster/main armhf nginx-full armhf 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
Err:15 http://raspbian.raspberrypi.org/raspbian buster/main armhf nginx all 1.14.2-2
  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/r/redis/redis-tools_5.0.3-4_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/r/redis/redis-server_5.0.3-4_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/nginx-common_1.14.2-2_all.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-auth-pam_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-dav-ext_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-echo_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-geoip_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-image-filter_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-subs-filter_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-upstream-fair_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-http-xslt-filter_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-mail_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/libnginx-mod-stream_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/nginx-full_1.14.2-2_armhf.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/n/nginx/nginx_1.14.2-2_all.deb  404  Not Found [IP: 2a00:1098:0:80:1000:75:0:3 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
^[[31m*** ----------------------------------^[(B^[[m
^[[31m*** ERROR ERROR ERROR ERROR ERROR ***^[(B^[[m
^[[31m
 ^[(B^[[m
^[[31m*** ----------------------------------^[(B^[[m
^[[31m*** See above lines for error message^[(B^[[m
^[[31m*** Setup NOT completed^[(B^[[m
^[[31m*** More information in the "install.log" file^[(B^[[m

Is anyone able to suggest what is the problem and how I can fix it? I am a complete Noob with this sort of thing.

Thanks!

**Comment by thorrak** [comment]
That's strange - those error messages would indicate that there potentially are issues with the servers that host Raspbian. Try running the following:

`sudo apt-get update`
`sudo apt-get upgrade`
`shutdown -r now`

This will then restart your Raspberry Pi - log back in and try installing Fermentrack again.

**Comment by suchidog** [comment]
Hey thorrak, 

Thanks for the reply. When i so the update I get:

pi@raspberrypi:~ $ sudo apt-get update
Hit:1 http://archive.raspberrypi.org/debian buster InRelease
Get:2 http://raspbian.raspberrypi.org/raspbian buster InRelease [15.0 kB]
Reading package lists... Done                                           
E: Repository 'http://raspbian.raspberrypi.org/raspbian buster InRelease' changed its 'Suite' value from 'testing' to 'stable'
N: This must be accepted explicitly before updates for this repository can be applied. See apt-secure(8) manpage for details.

**Comment by thorrak** [comment]
Yep, there's the issue! Apparently this is a problem with certain releases of Buster for Raspberry Pi 4s.

Discussion is here: https://www.raspberrypi.org/forums/viewtopic.php?f=91&t=245022

To fix it, run: `sudo apt update -y` and then re-run the commands I suggested before

**Comment by suchidog** [comment]
Thanks thorrak! I will try that now

**Comment by suchidog** [comment]
Thanks thorrak it worked

**Comment by suchidog** [comment]
thorrak I am just trying to install brew-tools now but get the following error:


pi@raspberrypi:~/brewpi-tools $ git clone https://github.com/thorrak/brewpi-tools
Cloning into 'brewpi-tools'...
remote: Enumerating objects: 504, done.
remote: Total 504 (delta 0), reused 0 (delta 0), pack-reused 504
Receiving objects: 100% (504/504), 138.82 KiB | 313.00 KiB/s, done.
Resolving deltas: 100% (305/305), done.
pi@raspberrypi:~/brewpi-tools $ sudo ./install.sh

Checking for Internet connection...
------------------------------------
Could not ping github.com. Are you sure you have a working Internet connection?
Installer will exit, because it needs to fetch code from github.com
pi@raspberrypi:~/brewpi-tools $ 

Is this just a Github issue?

**Comment by thorrak** [comment]
Possibly... Are you on a network that filters pings?

Try typing `ping -c 5 github.com` and see what happens. Then try `ping -c 5 yahoo.com`

**Comment by suchidog** [comment]
pi@raspberrypi:~ $ ping -c 5 github.com
PING github.com (13.236.229.21) 56(84) bytes of data.

--- github.com ping statistics ---
5 packets transmitted, 0 received, 100% packet loss, time 173ms

pi@raspberrypi:~ $ ping -c 5 yahoo.com
PING yahoo.com(media-router-fp1.prod1.media.vip.bf1.yahoo.com (2001:4998:58:1836::10)) 56 data bytes
64 bytes from media-router-fp1.prod1.media.vip.bf1.yahoo.com (2001:4998:58:1836::10): icmp_seq=1 ttl=45 time=248 ms
64 bytes from media-router-fp1.prod1.media.vip.bf1.yahoo.com (2001:4998:58:1836::10): icmp_seq=2 ttl=45 time=248 ms
64 bytes from media-router-fp1.prod1.media.vip.bf1.yahoo.com (2001:4998:58:1836::10): icmp_seq=3 ttl=45 time=247 ms
64 bytes from media-router-fp1.prod1.media.vip.bf1.yahoo.com (2001:4998:58:1836::10): icmp_seq=4 ttl=45 time=248 ms
64 bytes from media-router-fp1.prod1.media.vip.bf1.yahoo.com (2001:4998:58:1836::10): icmp_seq=5 ttl=45 time=247 ms

--- yahoo.com ping statistics ---
5 packets transmitted, 5 received, 0% packet loss, time 8ms
rtt min/avg/max/mdev = 246.919/247.497/248.180/0.749 ms


yes looks like I can't ping GitHub. Unsure why?

**Comment by thorrak** [comment]
Honestly, I'm at a complete loss. I searched, and found this comment which might be worth a try: https://github.com/Homebrew/brew/issues/1414#issuecomment-259034930

**Comment by suchidog** [comment]
Nope didn't work unfortunately. even tried a restart of the raspberrypi

**Comment by suchidog** [comment]
It seemed to work for the command prior on https (git clone https://github.com/thorrak/brewpi-tools)  but then perhaps the install.sh is coming from http? Is there a way to get it from the https?

On 5 Dec 2019, at 3:11 pm, John <notifications@github.com<mailto:notifications@github.com>> wrote:


Honestly, I'm at a complete loss. I searched, and found this comment which might be worth a try: Homebrew/brew#1414 (comment)<https://github.com/Homebrew/brew/issues/1414#issuecomment-259034930>

**Comment by lbussy** [comment]
@thorrak I thought you merged in the change that removed ping in favor of testing with curl?  Remember that?

**Comment by thorrak** [comment]
Gooooood point.  @suchidog - what command are you running to install Fermentrack?

**Comment by suchidog** [comment]
I think it was a curl command? I just followed the video instructions. The only issue I had on that install was the -y addition on the update.

Sent by my iPhone

On 5 Dec 2019, at 23:20, John <notifications@github.com> wrote:

﻿

Gooooood point. @suchidog<https://github.com/suchidog> - what command are you running to install Fermentrack?

**Comment by suchidog** [comment]
yeah it was curl -L install.fermentrack.com | sudo bash

It seems this ping issue is really widespread for users of GitHub. I have read multiple forum posts on it and tried their suggested solutions without success.

**Comment by thorrak** [comment]
Yeah - as @lbussy noted, there is a simple fix that apparently didn’t make it in for this. When I get back home I’ll try to get that in.

**Comment by suchidog** [comment]
awesome thanks!

On 6 Dec 2019, at 7:20 am, John <notifications@github.com<mailto:notifications@github.com>> wrote:


Yeah - as @lbussy<https://github.com/lbussy> noted, there is a simple fix that apparently didn’t make it in for this. When I get back home I’ll try to get that in.

**Comment by thorrak** [comment]
Hey @suchidog -- What _exactly_ are you doing that is causing this issue? I've dug through the code, and if you're just running the one-line install command `curl -L install.fermentrack.com | sudo bash` things should work, and you should be installing from the `fermentrack-tools` repo. 

From your error message though, it sounds like you're installing from the `brewpi-tools` repo. Are you trying to install classic BrewPi or Fermentrack?

**Comment by suchidog** [comment]
Hey @thorrak 

I am following this video https://www.youtube.com/watch?v=vUaPao_wBGI&list=PLCs4FqrNRHd0Br1XZfrIPrv9QU-tEcDRz&index=2

I can successfully run the command: git clone https://github.com/thorrak/brewpi-tools

However the next command:

sudo ./install.sh

cannot find an internet connection to Github

**Comment by thorrak** [comment]
Ah, got it. That is not a tutorial for Fermentrack, that's a tutorial for the classic BrewPi interface which uses a different set of tooling to install.

**Comment by suchidog** [comment]
sorry I should have opened a new chat under brew-tools

On 7 Dec 2019, at 7:23 am, John <notifications@github.com<mailto:notifications@github.com>> wrote:


Ah, got it. That is not a tutorial for Fermentrack, that's a tutorial for the classic BrewPi interface which uses a different set of tooling to install.

**Comment by thorrak** [comment]
No worries - Just means I need to fix it somewhere different. 👍 I'll get the fix drafted and will push it up. 

One thing to be aware of - you may be better off actually using @lbussy 's project [BrewPiRemix](https://www.brewpiremix.com/) if you're interested in the classic interface instead of Fermentrack. I can't guarantee that the instructions that you're following still work with modern versions of Rapsbian as there have been a lot of changes since they were drafted. @lbussy has worked to try to modernize the legacy interface of BrewPi so as to keep it working (and add a bunch of new features as well).

I'm still partial to Fermentrack, for obvious reasons, but his is a great project!

**Comment by suchidog** [comment]
Thanks very much thorrak

**Comment by suchidog** [comment]
Thanks very much.

Will I be able to see on the GitHub site once the update has been pushed?

Cheers,
Ben

On 7 Dec 2019, at 7:41 am, John <notifications@github.com<mailto:notifications@github.com>> wrote:


No worries - Just means I need to fix it somewhere different. 👍 I'll get the fix drafted and will push it up.

One thing to be aware of - you may be better off actually using @lbussy<https://github.com/lbussy> 's project BrewPiRemix<https://www.brewpiremix.com/> if you're interested in the classic interface instead of Fermentrack. I can't guarantee that the instructions that you're following still work with modern versions of Rapsbian as there have been a lot of changes since they were drafted. @lbussy<https://github.com/lbussy> has worked to try to modernize the legacy interface of BrewPi so as to keep it working (and add a bunch of new features as well).

I'm still partial to Fermentrack, for obvious reasons, but his is a great project!

**Comment by thorrak** [comment]
@suchidog The update to BrewPi-tools should already be up - https://github.com/thorrak/brewpi-tools

**Comment by suchidog** [comment]
Hey John,

Thanks very much for all your work.

Although I am confused!

I get to the 1 min 9 sec mark of your video https://www.youtube.com/watch?v=vUaPao_wBGI&list=PLCs4FqrNRHd0Br1XZfrIPrv9QU-tEcDRz&index=2

which I then can’t do the install command because of the internet test which I understanding you have fixed in your instructions https://github.com/thorrak/brewpi-tools/blob/master/install.sh

I am just not too sure how to interpret what to actually write into the terminal app. Are you able to assist? Sorry for being a dumbass!

Cheers,
Ben

On 16 Dec 2019, at 4:01 pm, John <notifications@github.com<mailto:notifications@github.com>> wrote:


@suchidog<https://github.com/suchidog> The update to BrewPi-tools should already be up - https://github.com/thorrak/brewpi-tools

**Comment by lbussy** [comment]
Basically, you are doing it wrong.   There are two projects: Fermentrack (the front end) and the ESP8266 port of BrewPi.  You can accomplish BOTH by following the instructions at [Fermentrack.com](http://fermentrack.com/).  

High level, however, you start the Fermentrack install process on your Raspberry Pi by issuing one command on the Pi:

```
curl -L install.fermentrack.com | sudo bash
```

Fermentrack will give you the option to flash your ESP8266 with the proper firmware.

**Comment by suchidog** [comment]
Thanks for that.

I actually already have fermentrack installed and running on my raspberry pi. The part I was having issue with was the brew-tools install (as I couldn’t ping the GitHub server as per the video instructions @ 1.08 https://www.youtube.com/watch?v=vUaPao_wBGI&list=PLCs4FqrNRHd0Br1XZfrIPrv9QU-tEcDRz&index=2).

Are you saying that I can now flash the BrewPiLess and run BrewPiLess from inside the Fermentrack interface rather than install via terminal?

Thanks,
Ben

On 17 Dec 2019, at 5:04 am, Lee Bussy <notifications@github.com<mailto:notifications@github.com>> wrote:


Basically, you are doing it wrong. There are two projects: Fermentrack (the front end) and the ESP8266 port of BrewPi. You can accomplish BOTH by following the instructions at Fermentrack.com<http://fermentrack.com/>.

High level, however, you start the Fermentrack install process on your Raspberry Pi by issuing one command on the Pi:

curl -L install.fermentrack.com<http://install.fermentrack.com> | sudo bash


Fermentrack will give you the option to flash your ESP8266 with the proper firmware.

**Comment by lbussy** [comment]
> Are you saying that I can now flash the BrewPiLess and run BrewPiLess from inside the Fermentrack interface rather than install via terminal?

BrewPiLess is yet another project.  Fermentrack will flash the ESP8266 BrewPi, or the Arduino Uno BrewPi from within the interface.

**Comment by suchidog** [comment]
Thanks Lee.

Sorry I am using incorrect names.

To be clear I don’t need to do any of the things on this video https://www.youtube.com/watch?v=vUaPao_wBGI&list=PLCs4FqrNRHd0Br1XZfrIPrv9QU-tEcDRz&index=2 to install brewpi? It can all be done within the interface now?

On 17 Dec 2019, at 8:42 am, Lee Bussy <notifications@github.com<mailto:notifications@github.com>> wrote:


Are you saying that I can now flash the BrewPiLess and run BrewPiLess from inside the Fermentrack interface rather than install via terminal?

BrewPiLess is yet another project. Fermentrack will flash the ESP8266 BrewPi, or the Arduino Uno BrewPi from within the interface.

**Comment by lbussy** [comment]
Yes, follow the instructions that prompt you.

**Comment by suchidog** [comment]
thanks very much

On 17 Dec 2019, at 9:58 am, Lee Bussy <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes, follow the instructions that prompt you.
